tests/rptest/redpanda_cloud_tests/omb_validation_test.py

```diff
@@ -0,0 +1,437 @@
+
+    def _consumer_count(self, egress_rate) -> int:
+        return max(egress_rate // (4 * MiB), 8)
+
+    def _mb_to_mib(self, mb):
+        return math.floor(0.9537 * mb)
+
+    @cluster(num_nodes=12)
+    def test_max_connections(self):
+        tier_config = self.redpanda.advertised_tier_config
+
+        # Constants
+        #
+
+        PRODUCER_TIMEOUT_MS = 5000
+        OMB_WORKERS = 4
+        SWARM_WORKERS = 7
+
+        # OMB parameters
+        #
+
+        producer_rate = tier_config.ingress_rate // 5
+        subscriptions = max(
+            tier_config.egress_rate // tier_config.ingress_rate, 1)
+        total_producers = self._producer_count(producer_rate)
+        total_consumers = self._consumer_count(producer_rate * subscriptions)
+        warmup_duration = 1  # minutes
+        test_duration = 5  # minutes
+
+        workload = {
+            "name": "MaxConnectionsTestWorkload",
+            "topics": 1,
+            "partitions_per_topic": self._partition_count(),
+            "subscriptions_per_topic": subscriptions,
+            "consumer_per_subscription": max(total_consumers // subscriptions,
+                                             1),
+            "producers_per_topic": total_producers,
+            "producer_rate": producer_rate // (1 * KiB),
+            "message_size": 1 * KiB,
+            "payload_file": "payload/payload-1Kb.data",
+            "consumer_backlog_size_GB": 0,
+            "test_duration_minutes": test_duration,
+            "warmup_duration_minutes": warmup_duration,
+        }
+
+        driver = {
+            "name": "MaxConnectionsTestDriver",
+            "replication_factor": 3,
+            "request_timeout": 300000,
+            "producer_config": {
+                "enable.idempotence": "true",
+                "acks": "all",
+                "linger.ms": 1,
+                "max.in.flight.requests.per.connection": 5,
+            },
+            "consumer_config": {
+                "auto.offset.reset": "earliest",
+                "enable.auto.commit": "false",
+            },
+        }
+
+        validator = self.base_validator | {
+            OMBSampleConfigurations.AVG_THROUGHPUT_MBPS: [
+                OMBSampleConfigurations.gte(
+                    self._mb_to_mib(producer_rate // (1 * MB))),
+            ],
+        }
+
+        # ProducerSwarm parameters
+        #
+
+        producer_kwargs = {}
+        producer_kwargs['min_record_size'] = 64
+        producer_kwargs['max_record_size'] = 64
+
+        effective_msg_size = producer_kwargs['min_record_size'] + (
+            producer_kwargs['max_record_size'] -
+            producer_kwargs['min_record_size']) // 2
+
+        conn_limit = tier_config.connections_limit - 3 * (total_producers +
+                                                          total_consumers)
+        _target_per_node = conn_limit // SWARM_WORKERS
+        _conn_per_node = int(_target_per_node * 0.7)
```

Comment:
Yeah this bit maybe needs a bit of improving before merging:

 - At least a comment explaining where the magic factor comes from
 - As discussed probably better to calculate the bootstrap connections based on the clients/brokers. Especially as this can vary quite a bit white node count. For a 3 broker cluster each client will open ~33% percent more connections than expected while for a 9 node cluster it's only ~11%.

Alternatively what I did in manual testing is just to resolve the seed url to the actual server URLs and then pass those directly. But that might be difficult in DT.